src/main/java/com/fasterxml/jackson/dataformat/xml/deser/XmlDeserializationContext.java

```diff
@@ -97,13 +98,19 @@ public Object readRootValue(JsonParser p, JavaType valueType,
     {
         // 18-Sep-2021, tatu: Complicated mess; with 2.12, had [dataformat-xml#374]
         //    to disable handling. With 2.13, via [dataformat-xml#485] undid this change
-        if (_config.useRootWrapping()) {
-            return _unwrapAndDeserialize(p, valueType, deser, valueToUpdate);
-        }
-        if (valueToUpdate == null) {
-            return deser.deserialize(p, this);
+        try {
+            if (_config.useRootWrapping()) {
+                return _unwrapAndDeserialize(p, valueType, deser, valueToUpdate);
+            }
+            if (valueToUpdate == null) {
+                return deser.deserialize(p, this);
+            }
+            return deser.deserialize(p, this, valueToUpdate);
+        } catch (IndexOutOfBoundsException e) {
+            // If value is invalid without end character, the deserialize will
+            //     read pass the array bound and throws IndexOutOfBoundException
+            throw new JsonParseException(p, "Invalid value with missing JsonToken.END_OBJECT.", e);
```

Comment:
No, no, no. We should not be catching `IndexOutOfBoundsException`s; there are indications of issues somewhere else. I am not a big fan of defensive programming in this style unless there's absolutely no other choice (for 3rd party deps)